komorebi-wallpaper-creator.in

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python3
+
+import argparse
+import gi
+import logging
+import sys
+import os
+
+gi.require_versions({
+	'Clutter': '1.0',
+	'Gtk': '3.0'
+})
+
+from gi.repository import Gtk, Gio
+
+import komorebi.consts
+
+from komorebi.wallpaper_creator.window import WallpaperWindow
+
+def main_parser():
+	parser = argparse.ArgumentParser()
+	parser.add_argument('-v', '--version',
+						action='store_true',
+						help='show current version')
+	parser.add_argument('-l', '--log',
+						type=str,
+						choices=['NORMAL', 'INFO', 'DEBUG'],
+						default='NORMAL',
+						help="set logging level for komorebi")
+
+	return parser
+
+def main():
+	print(f'Welcome to {komorebi.consts.package_name} Wallpaper Creator')
+
+	parser = main_parser()
+	args = parser.parse_args()
+
+	if args.version:
+		print(f'Version: {komorebi.consts.package_version}\nMaintained by: Komorebi Team\n')
+		return
+	
+	log_level = logging.WARNING
+	log_format = ''
+	
+	if args.log:
+		if args.log == 'INFO':
+			log_level = logging.INFO
+			log_format = '[%(levelname)s]: %(message)s'
+		elif args.log == 'DEBUG':
+			log_level = logging.DEBUG
+			log_format = '[%(levelname)s] (%(asctime)s): %(message)s'
+	
+	logging.basicConfig(format=log_format, level=log_level, datefmt='%H:%M:%S')
+
+	# Load resources
+	resource_path = os.path.join(komorebi.consts.package_datadir, 'komorebi.gresource')
+	resource = Gio.Resource.load(resource_path)
+	resource._register()
+
+	Gtk.init(sys.argv)
+	logging.debug('Gtk initialized')
+
+	window = WallpaperWindow()
+
+	main_settings = Gtk.Settings.get_default()
+	main_settings.props.gtk_application_prefer_dark_theme = True
+	main_settings.props.gtk_xft_antialias = 1
+	main_settings.props.gtk_xft_rgba = "none"
+	main_settings.props.gtk_xft_hintstyle = "slight"
+
+	Gtk.main()
+
+if __name__ == '__main__':
+	main()
```

Comment:
Please no, editorconfig is awful. It is okay to use in other languages where no better tooling is available. Just use flake8 here.

Reply:
> Please no, editorconfig is awful. It is okay to use in other languages where no better tooling is available. Just use flake8 here.

I just mentioned `.editorconfig` because it can be generalized to a lot of files, like `meson.build`, `.json`, pretty much everything and a lot of editors parse it natively and put the newline at the end for you.

Reply:
Fixed

Reply:
> I just mentioned .editorconfig because it can be generalized to a lot of files, like meson.build, .json, pretty much everything and a lot of editors parse it natively and put the newline at the end for you.

I agree. You can have other tools for pyhton files but it's nice to have all files following a standard. You can configure an `.editconfig` file to interact with specific files or paths leaving the project much more "sanitized".